src/cli/circom.rs

```diff
@@ -0,0 +1,124 @@
+#[cfg(not(target_arch = "wasm32"))]
+pub mod non_wasm {
+    use std::{
+        env,
+        fs::{self, File},
+        io::Write,
+        os::unix::prelude::PermissionsExt,
+        path::Path,
+        process::{exit, Command},
+    };
+
+    use ansi_term::Colour::{Green, Red};
+    use anyhow::{bail, Result};
+    use camino::Utf8PathBuf;
+
+    use crate::cli::paths::{circom_binary_path, circom_dir};
+
+    const CIRCOM_VERSION: &str = "2.1.6";
+
+    fn download_circom_binary(path: impl AsRef<Path>) -> Result<Command> {
+        let url = match env::consts::OS {
+            "linux" => {
+                "https://github.com/iden3/circom/releases/download/v2.1.6/circom-linux-amd64"
+            }
+            "macos" => {
+                "https://github.com/iden3/circom/releases/download/v2.1.6/circom-macos-amd64"
+            }
+            "windows" => {
+                "https://github.com/iden3/circom/releases/download/v2.1.6/circom-windows-amd64.exe"
+            }
+            _ => {
+                eprintln!("Unsupported OS");
+                exit(1);
+            }
+        };
+
+        let response = reqwest::blocking::get(url)?.bytes()?;
+        let mut out = File::create(path.as_ref())?;
+        out.write_all(&response)?;
+
+        #[cfg(unix)]
+        fs::set_permissions(path.as_ref(), fs::Permissions::from_mode(0o755))?;
+
+        Ok(Command::new(path.as_ref().as_os_str()))
+    }
+
+    /// We try the following places to find `circom`, in this order
+    ///  1. `LURK_CIRCOM_PATH`
+    ///  2. `.lurk/circom/circom`
+    ///
+    /// We *do not* consider the case where the user already has some
+    /// `circom` binary downloaded. The user will have two possibly
+    /// conflicting circom binaries floating around. However, things
+    /// should be kept separate as Lurk will never touch the user binary
+    /// and the user should never manually call the Lurk Circom binary.
```

Comment:
Perhaps this could be configurable so that the user isn't forced to have duplicate binaries? One idea would be to list the `circom` binary as a pre-requisite in the docs, search for the `CIRCOM` env var or its default location when running the CLI, and if it can't be found only then download the binary to `LURK_CIRCOM_PATH`, or simply return an error.